app/models/energy/construction.py

```diff
@@ -832,6 +23,9 @@ class EnergyConstructionWindow(BaseModel):
         max_length=100
     )
 
+    layers: List[constr(regex=r'^[\s.A-Za-z0-9_-]*$',
+                        min_length=1, max_length=100)]
+
     materials: List[
```

Comment:
Could we just make materials optional?  Then the "abridged version" is just a construction without that optional field and an "extended version" is a construction with the optional field. There could be a convention that if materials are present, layers must refer to materials nested within the object.

Reply:
Materials are required to recreate a object and they need to be required on the non-abridged version of the object.  See my reply below for a longer description.